filebeat/prospector/prospector.go

```diff
@@ -105,6 +105,10 @@ func (p *Prospector) Run() {
 				logp.Info("Prospector channel stopped")
 				return
 			case event := <-p.harvesterChan:
+				// Add ttl if cleanOlder is enabled
+				if p.config.CleanOlder != 0 {
+					event.FileState.Ttl = int64(p.config.CleanOlder.Seconds())
```

Comment:
Have to read up if/when/how FileState is shared between multiple go-routines. But in case it's shared, this is not safe. Would be better to set the TTL when FileState is generated (as it's not supposed to change).